src/NuGet.Clients/NuGet.CommandLine/Common/SelfUpdater.cs

```diff
@@ -59,96 +58,95 @@ public string AssemblyLocation
             }
         }
 
-        public void UpdateSelf(bool prerelease)
+        public async Task UpdateSelfAsync(bool prerelease, string updateFeed)
         {
             Assembly assembly = typeof(SelfUpdater).Assembly;
-            CoreV2.NuGet.SemanticVersion version = GetNuGetVersion(assembly) ?? new CoreV2.NuGet.SemanticVersion(assembly.GetName().Version);
-            SelfUpdate(AssemblyLocation, prerelease, version);
+            var version = GetNuGetVersion(assembly) ?? new NuGetVersion(assembly.GetName().Version);
+            await SelfUpdateAsync(AssemblyLocation, prerelease, version, updateFeed, CancellationToken.None);
```

Comment:
This `UpdateSelf`/`SelfUpdate` naming is pretty confusing. Is this a good time to improve it?

Reply:
What do you have in mind?

Reply:
`SelfUpdate` => `UpdateSelfFromVersionAsync` maybe?

Reply:
Ahhh, makes sense. i'll change it.